### ISSUE TITLE
Generated links not working.

### ISSUE THREAD

**Issue by Didas-git** [HiDeoo/starlight-typedoc#35]
### Describe the bug

The generated links by typedoc have the interfaces/types/etc... in the casing they are present in the source code and also have the extension (.md), meanwhile in starlight the paths are all normalized (lower cased) and do not have the .md extension present, this makes it so the links just do not work and redirect to 404.
I have also noticed that if `typeDoc.publicPath` is not the same as the `output` the links will redirect to the base domain which also results in 404.

### To Reproduce

1- have the following:
```ts
// A.ts
import type { B } from "./B";

export interface A {
  a: string;
  b: B
}

// B.ts
export type B = number | bigint
```

2- Use `createStarlightTypeDocPlugin()` and pass `A.ts` as your entry point to function returned.

### Expected behavior

I would expect the links to just work but they dont and i havent found any work around.

### How often does this bug happen?

Every time

### System Info

- OS - Linux 5.15.146.1-microsoft-standard-WSL2 x86_64 x86_64
- Browser - Google Chrome 123.0.6312.106 (Official Build) (64-bit) (cohort: Stable)
- Astro version - 4.5.16
- Starlight version - 0.21.4
- Starlight Typedoc version - 0.10.0
- Typedoc version - 0.25.12
- Typedoc Markdown version - 4.0.0-next-54

**Comment by HiDeoo** [comment]
Thanks for your report :raised_hands:

Would you be able to provide a small repro with the steps you provided above? I tried reproducing using the content you provided and the following configuration which matches your description:

```js
import starlight from '@astrojs/starlight'
import { defineConfig } from 'astro/config'
import { createStarlightTypeDocPlugin } from 'starlight-typedoc'

const [starlightTypeDoc, typeDocSidebarGroup] = createStarlightTypeDocPlugin()

export default defineConfig({
  integrations: [
    starlight({
      plugins: [
        starlightTypeDoc({
          entryPoints: ['../fixtures/test/A.ts'],
          tsconfig: '../fixtures/test/tsconfig.json',
        }),
      ],
      sidebar: [typeDocSidebarGroup],
      title: 'Starlight TypeDoc Example',
    }),
  ],
})
```

Unfortunately, I wasn't able to reproduce the issue.

**Comment by Didas-git** [comment]
My best guess is that it might be due to some tsconfig option.
Unfortunately im not able to provide a simple reproduction at this time, but i will leave the link of my repository thats causing the issue.
https://github.com/Didas-git/lilybird/commit/0536a9eee9b4c63fa2d1275daec1a209da09bfe7

I can try to narrow it down at a later date if thats better 🙏

**Comment by Didas-git** [comment]
I made a smaller reproduction of this issue here: https://github.com/Didas-git/typedoc-linking-repro/tree/main/packages/docs
And here some screenshots to illustrate the problem
Sidebar has correct redirect:
![image](https://github.com/HiDeoo/starlight-typedoc/assets/80978939/30521a73-fa71-44f3-971e-2d34790d98b3)
Links do not have the correct redirect:
![image](https://github.com/HiDeoo/starlight-typedoc/assets/80978939/dc441e8a-cc57-4f43-90ae-8f2b6d9046c5)

**Comment by mProjectsCode** [comment]
I am facing the same issue on [this site](https://www.moritzjung.dev/obsidian-js-engine-plugin-docs/api/classes/api/).

**Comment by lajbel** [comment]
+1

**Comment by HiDeoo** [comment]
Thanks for your reproduction, really helped a lot :raised_hands:

I finally tracked down the issue to breaking changes in `typedoc-plugin-markdown`. This should now be fixed in the version [0.11.0](https://github.com/HiDeoo/starlight-typedoc/releases/tag/v0.11.0) that I just published and now requires at least the version `4.0.0-next.54` of `typedoc-plugin-markdown`.

**Comment by mProjectsCode** [comment]
Thank you so much 🎉
